community/io/src/test/java/org/neo4j/io/fs/watcher/DefaultFileSystemWatcherTest.java

```diff
@@ -0,0 +1,328 @@
+
+        TestWatchEvent<String> event = new TestWatchEvent( ENTRY_MODIFY, null );
+        TestWatchKey watchKey = new TestWatchKey( asList( event ) );
+
+        prepareWatcher( watchKey );
+
+        watch( watcher );
+
+        listener.assertNoEvents();
+    }
+
+    private void prepareWatcher( TestWatchKey watchKey ) throws InterruptedException
+    {
+        when( watchServiceMock.take() ).thenReturn( watchKey )
+                .thenAnswer( new ThrowsExceptionClass( InterruptedException.class ) );
+    }
+
+    private void watch( TestFileSystemWatcher watcher )
+    {
+        try
+        {
+            watcher.startWatching();
+        }
+        catch ( InterruptedException ignored )
+        {
+            // expected
+        }
+    }
+
+    private TestFileSystemWatcher createWatcher()
+    {
+        return new TestFileSystemWatcher( watchServiceMock );
+    }
+
+    private static class TestFileSystemWatcher extends DefaultFileSystemWatcher
+    {
+
+        private boolean closed;
+
+        TestFileSystemWatcher( WatchService watchService )
+        {
+            super( watchService );
+        }
+
+        @Override
+        public void close() throws IOException
+        {
+            super.close();
+            closed = true;
+        }
+
+        public boolean isClosed()
+        {
+            return closed;
+        }
+    }
+
+    private static class TestWatchKey implements WatchKey
+    {
+        private List<WatchEvent<?>> events;
+        private boolean canceled;
+
+        TestWatchKey( List<WatchEvent<?>> events )
+        {
+            this.events = events;
+        }
+
+        @Override
+        public boolean isValid()
+        {
+            return false;
+        }
+
+        @Override
+        public List<WatchEvent<?>> pollEvents()
+        {
+            return events;
+        }
+
+        @Override
+        public boolean reset()
+        {
+            return false;
+        }
+
+        @Override
+        public void cancel()
+        {
+            canceled = true;
+        }
+
+        @Override
+        public Watchable watchable()
+        {
+            return null;
+        }
+
+        public boolean isCanceled()
+        {
+            return canceled;
+        }
+    }
+
+    private static class TestWatchEvent<T> implements WatchEvent
+    {
+
+        private Kind<T> eventKind;
+        private T fileName;
+
+        TestWatchEvent( Kind<T> eventKind, T fileName )
+        {
+            this.eventKind = eventKind;
+            this.fileName = fileName;
+        }
+
+        @Override
+        public Kind kind()
+        {
+            return eventKind;
+        }
+
+        @Override
+        public int count()
+        {
+            return 0;
+        }
+
+        @Override
+        public T context()
+        {
+            return fileName;
+        }
+    }
+
+    private static class AssertableFileEventListener extends FileWatchEventListenerAdapter {
+
```

Comment:
Most of your classes I've seen start and sometimes end with a blank line, any chance to remedy that in your editor config?